tests/unit/tree/request_node/test_run.py

```diff
@@ -62,6 +62,91 @@ def test_calls_request(self, mock_session, mock_time_sleep):
             "options": {"timeout": 2.3, "verify": False},
         }
 
+    @mark.context("when has no `Content-Type` header")
+    @mark.it("should use `json` parameter to send body")
+    def test_content_type_is_not_defined(self, mock_session):
+        request = RequestNode(
+            {
+                "path": "http://foo.com",
+                "name": "request_name",
+            },
+            endpoint=EndpointNode(
+                {
+                    "name": "endpoint_name",
+                    "requests": [{}],
+                }
+            ),
+        )
+
+        request.run()
+
+        mock_session().request.assert_called_once_with(
+            request.http_method,
+            request.full_url_path,
+            headers=request.headers,
+            params=request.params,
+            json=request.body,
+            allow_redirects=False,
+        )
+
+    @mark.context("when `Content-Type` header is `application/json`")
+    @mark.it("should use `json` parameter to send body")
+    def test_content_type_is_application_json(self, mock_session):
+        request = RequestNode(
+            {
+                "path": "http://foo.com",
+                "name": "request_name",
+                "headers": {"content-type": "application/json"},
+            },
+            endpoint=EndpointNode(
+                {
+                    "name": "endpoint_name",
+                    "requests": [{}],
+                }
+            ),
+        )
+
+        request.run()
+
+        mock_session().request.assert_called_once_with(
+            request.http_method,
+            request.full_url_path,
+            headers=request.headers,
+            params=request.params,
+            json=request.body,
+            allow_redirects=False,
+        )
+
+    @mark.context("when `Content-Type` header isn't `application/json`")
+    @mark.it("should use `data` parameter to send `body`")
+    def test_content_type_is_not_application_json(self, mock_session):
+        request = RequestNode(
+            {
+                "path": "http://foo.com",
+                "name": "request_name",
+                "headers": {
+                    "content-type": "application/x-www-form-urlencoded"
+                },
+            },
+            endpoint=EndpointNode(
+                {
+                    "name": "endpoint_name",
+                    "requests": [{}],
+                }
+            ),
+        )
+
+        request.run()
+
+        mock_session().request.assert_called_once_with(
+            request.http_method,
+            request.full_url_path,
+            headers=request.headers,
+            params=request.params,
+            data=request.body,
+            allow_redirects=False,
+        )
+
```

Comment:
Maybe we can add more (random) headers to prove that the feature works with many?

Reply:
great! i will do it tonight

Reply:
Sure! I would simply add some more headers in `test_content_type_is_not_application_json`:

```python
        request = RequestNode(
            {
                "path": "http://foo.com",
                "name": "request_name",
                "headers": {
                    "authorization": "Bearer token",
                    "content-type": content_type,
                    "X-Request-ID": "some_id",
                },
            },
            endpoint=EndpointNode(
                {
                    "name": "endpoint_name",
                    "requests": [{}],
                }
            ),
        )
```

And same in `test_content_type_is_application_json`.

This would show that the feature works when `Content-Type` is "hidden" among other headers.

Reply:
Sure! I would simply add some more headers in `test_content_type_is_not_application_json`:

```python
        request = RequestNode(
            {
                "path": "http://foo.com",
                "name": "request_name",
                "headers": {
                    "authorization": "Bearer token",
                    "content-type": content_type,
                    "X-Request-ID": "some_id",
                },
            },
            endpoint=EndpointNode(
                {
                    "name": "endpoint_name",
                    "requests": [{}],
                }
            ),
        )
```

And same in `test_content_type_is_application_json`.

This would show that the feature works when `Content-Type` is "hidden" among other headers.

Reply:
Sure! i will do it, thanks! 😄

Reply:
@Crocmagnon done! 👀